usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js

```diff
@@ -8,119 +8,348 @@
+                that.roomOccupantListID
+            ];                
+            that.room = {};
             that.room = {
-                "id": that.entityProperties.id, 
-                "name": that.entityProperties.name,
-                "eventList": [],
-                "scheduleEntityID": that.entityProperties.parentID
+                "eventList": []
             };
-            Entities.editEntity(that.room.scheduleEntityID, {text: ""});
-            that.updateSignColor(that.room.id, [that.room.name, "GREEN"]);
-        } else {
+            that.request = Script.require('https://hifi-content.s3.amazonaws.com/brosche/dev/googleCalendar/resources/modules/request.js').request;
+            that.clearEventList(that.entityID);
+            that.updateSignColor(that.roomEntityIDs[1], ["GREEN"]);
+            that.updateSignColor(that.roomEntityIDs[2], ["GREEN"]);
+            that.googleRequest(that.token);
+        } else if (that.entityID === that.roomOccupantListID) {
+            that.room = {};
             that.room = {
-                "id": that.entityProperties.id, 
                 "occupants": []
             };
-            return;
+            Entities.editEntity(that.entityID, {
+                text: 'loading'
+            });
+        }
+    };
+
+
+    this.refreshToken = function(id) {
+        if (that.entityID === id) {
+            if (that.interval) {
+                console.log("CLEARING INTERVAL");
+                Script.clearInterval(that.interval);
+                that.interval = false;
+            }
+            that.entityProperties = Entities.getEntityProperties(that.entityID, ['userData']);
+            if (that.entityProperties.userData.length !== 0) {
+                try {
+                    that.userData = JSON.parse(that.entityProperties.userData);
+                } catch (e) {
+                    console.log(e, "no userData found");
+                    return;
+                }
+                that.token = that.userData.token;
+                that.expireTime = that.userData.expireTime;
+                that.sentAlready = false;
+                that.googleRequest(that.token);
+            }
+        }
+    };
+
+
+    this.googleRequest = function(token) {
+        var tomorrowMidnight = new Date();
+        tomorrowMidnight.setHours(HOURS_PER_DAY, 0, 0, 0);
+        that.scheduleURL = "https://www.googleapis.com/calendar/v3/calendars/" + 
+            that.calendarID + 
+            "/events?&timeMin=" + (new Date()).toISOString() +
+            "&timeMax=" + tomorrowMidnight.toISOString() +
+            "&showDeleted=" + false + 
+            "&singleEvents=" + true +
+            "&maxResults=2" +
+            "&orderBy=startTime" +
+            "&access_token=" + token;
+        if (!that.interval) {
+            that.interval = Script.setInterval(function() {
+                if ((new Date()).valueOf() > (that.expireTime - EXPIRY_BUFFER) && !that.sentAlready) {
```

Comment:
it looks like you are duplicating this code down on line 169? please pull both instances of this code into a new function, and have the interval call that function upon expiry